src/contract.ts

```diff
@@ -0,0 +1,78 @@
+import { provider } from './config';
+
+import * as BASE_REGISTRAR_BYTECODE from './assets/ENSRegistrar_bc.json';
+import * as NAME_WRAPPER_BYTECODE from './assets/NameWrapper_bc.json';
+import { ethers } from 'ethers';
+import { Version } from './metadata';
+
+
+export interface BaseError {}
+export class BaseError extends Error {
+  __proto__: Error;
+  constructor(message?: string) {
+    const trueProto = new.target.prototype;
+    super(message);
+
+    this.__proto__ = trueProto;
+  }
+}
+
+export interface ContractMismatchError {}
+export class ContractMismatchError extends BaseError {}
+
+export interface ContractNotFoundError {}
+export class ContractNotFoundError extends BaseError {}
+
+export interface OwnerNotFoundError {}
+export class OwnerNotFoundError extends BaseError {}
+
+
+export async function checkContract(
+  contractAddress: string,
+  tokenId: string
+): Promise<Version> {
+  let nftOwner;
+  const contract_code = await provider.getCode(contractAddress);
+  if (contract_code === '0x') {
+    throw new ContractNotFoundError(`${contractAddress} is not a contract`);
+  }
+  // TODO check hexId labelhash or intId namehash
+
+  try {
+    const contract = new ethers.Contract(
+      contractAddress,
+      ['function ownerOf(uint256 tokenId) public view returns (address)'],
+      provider
+    );
+    nftOwner = await contract.ownerOf(tokenId);
+  } catch (error) {
+    console.log('err', error.message);
+    throw new ContractMismatchError(
+      `${contractAddress} does not match with any ENS related contract`
+    );
+  }
+
+  if (nftOwner === '0x') {
```

Comment:
Does this ever happen? I think the contract reverts if the name owner does not exist.

<img width="1141" alt="Screenshot 2021-08-20 at 15 29 48" src="https://user-images.githubusercontent.com/2630/130248964-8074fe4c-6ee0-425a-91a1-3ad56eebc307.png">

Reply:
Unlike BaseRegistrarImplementation, it happens in NameWrapper.
<img width="1366" alt="image" src="https://user-images.githubusercontent.com/2774845/130277277-740620c9-116c-4c29-9d84-983829192d85.png">